### PR TITLE
Use List Initializations for Vectors.

### PR DESCRIPTION
**Pull request by AnilBK**
With [#50284](https://github.com/godotengine/godot/pull/50493), we now have list initialization for Vectors.

This PR doesn't replace all occurrences( For e.g: Vector push_backs inside a switch statement, or vector assignments after Vector clear()).

### CODE REVIEW

File: core/core_bind.cpp [hunk 1]
```diff
@@ -698,10 +698,7 @@ Variant Geometry2D::line_intersects_line(const Vector2 &p_from_a, const Vector2
 Vector<Vector2> Geometry2D::get_closest_points_between_segments(const Vector2 &p1, const Vector2 &q1, const Vector2 &p2, const Vector2 &q2) {
 	Vector2 r1, r2;
 	::Geometry2D::get_closest_points_between_segments(p1, q1, p2, q2, r1, r2);
-	Vector<Vector2> r;
-	r.resize(2);
-	r.set(0, r1);
-	r.set(1, r2);
+	Vector<Vector2> r = { r1, r2 };
 	return r;
```
Comment (timothyqiu) [inline core/core_bind.cpp:702]: In cases like this, `return {r1, r2};` also works. But I'm not sure if it's against our coding style...

**Comment by AnilBK** [comment]
> One thing we need to define style-wise is whether we want to enforce a trailing comma on multi-line list initializations, like we try to do for enum definitions and some other existing list initializations.
https://github.com/godotengine/godot/blob/a095c4bf65c821d926a0eeae42e8bfeb02b48b23/scene/animation/tween.h#L64-L96

For example: here,in a single file there is both trailing comma and not trailing comma.
 
> Personally I'm in favor of it as it makes things more consistent, and makes it easy to add new items when relevant. But I don't have a very strong opinion and I'm also fine with keeping this as is if others prefer it so.

I guess,we can leave it as it and enforce a definite style across everywhere once we have it.

 `internal ranlib command failed`
Also,how to fix macOS CI?
Will restarting CI by force pushing fix it?

File: core/core_bind.cpp (hunk 1 above)
Comment (AnilBK) [reply core/core_bind.cpp:702]: There is another case for Astar file as well.
Kept it that way to maintain readability.

**Review by AaronRecord — APPROVED** [review]

**Comment by akien-mga** [comment]
> For example: here,in a single file there is both trailing comma and not trailing comma.

Yes, because `_MAX` enum always mark the end of the enum, so there will never be any addition after that. But for other enums with `_MAX`, they should end with a trailing comma.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**